tests/e2e/hcp_machine_pool_test.go

```diff
@@ -26,54 +103,244 @@ var _ = Describe("Create Machine Pool", labels.Feature.Machinepool, func() {
-			mpName = "mp-72195-nsg"
-			_, err = rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
-				"--replicas", "1",
+	DescribeTable("Scale up/down a machine pool with invalid replica", labels.Critical, labels.Runtime.Day2,
+		func(instanceType string, updatedReplicas string, expectedErrMsg string) {
+			By("Create machinepool with instance " + instanceType)
+			mpName := common.GenerateRandomName("mp-60278", 2)
+			desiredReplicas := 1
+			_, err := rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
+				"--instance-type", instanceType,
+				"--replicas", fmt.Sprintf("%v", desiredReplicas),
 				"-y",
 			)
 			Expect(err).ToNot(HaveOccurred())
 			defer rosaClient.MachinePool.DeleteMachinePool(clusterID, mpName)
-			By("Check the machinepool detail by describe")
-			mpDescription, err = rosaClient.MachinePool.DescribeAndReflectNodePool(clusterID, mpName)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(mpDescription.AdditionalSecurityGroupIDs).To(BeEmpty())
+
+			_, err = rosaClient.MachinePool.EditMachinePool(clusterID, mpName,
+				"--replicas", fmt.Sprintf("%v", updatedReplicas),
+				"-y",
+			)
+			Expect(err.Error()).Should(ContainSubstring(expectedErrMsg))
+		},
+
+		Entry("Scale replica too large [id:60278]", constants.M52XLarge, "1000", "exceeds the maximum allowed"),
+		Entry("Scale replica to -1 [id:60278]", constants.M52XLarge, "-1", "Replicas must be a non-negative number"),
+		Entry("Scale replica to a char [id:60278]", constants.M52XLarge, "a", "invalid syntax"),
+	)
+
+	Describe("Scale up/down a machine pool enabling autoscale", func() {
+		When("with the valid min replica and max replica", func() {
+			It("should succeed to scale up/down the machine pool[id:60278]", labels.Medium, labels.Runtime.Day2, func() {
+				instanceType := constants.M52XLarge
+				By("Create machinepool with " + " instance " + instanceType + " and enable autoscale")
+
+				mpPrefix := "autoscale"
+				mpName := common.GenerateRandomName(mpPrefix, 2)
+				minReplica := 1
+				maxReplica := 3
+				_, err := rosaClient.MachinePool.CreateMachinePool(clusterID, mpName,
+					"--instance-type", instanceType,
+					"--enable-autoscaling",
+					"--min-replicas", fmt.Sprintf("%v", minReplica),
+					"--max-replicas", fmt.Sprintf("%v", maxReplica),
+					"-y",
+				)
+				Expect(err).ToNot(HaveOccurred())
+				defer rosaClient.MachinePool.DeleteMachinePool(clusterID, mpName)
+
+				// TODO There's issue here https://issues.redhat.com/browse/OCM-10147
+				//By("Scale a machine pool with unchanged parameters")
+				//scaleAutoScaledNodePool(mpName, "unchanged", minReplica, maxReplica, true)
+
+				By("Scale up a machine pool replicas from 1~3 to 2~5")
+				upMinReplica := 2
+				upMaxReplica := 5
+				scaleAutoScaledNodePool(mpName, "up", upMinReplica, upMaxReplica, true)
+
+				// Don't check the current replicas when scale down, because it won't change after reducing min_replica
```

Comment:
Is it a bug?

Reply:
I confirmed with DEV adecorte. He said it's as expected.

Reply:
If it is expected, no need to clarify here. Let's clarify in the case in Polarion.

Reply:
`It` spec already indicated the testing purpose. Too many layers will make the code hard to maintain.